### ISSUE TITLE
Update MIG/MIG_template Provider Constraints to a newer version than 7.X

### ISSUE THREAD

**Issue by pickmylight** [terraform-google-modules/terraform-google-github-actions-runners#165]
### TL;DR

As mentioned in #123:

This is still an issue today. We stumbled upon this problem with the mig & mag-template dependency and had to downgrade our complete googleProvider to <5.0.0. What's the Point sticking with mig ~>7.0? There were 4 major updates since (11.x.x)

### Expected behavior

When using a google Provider I expect to use the newest google provider versions and not being restricted by a sub-sub-subdependency inside of the VM Templating.

### Observed behavior

When installing with other Providers and modules there  ist a Dependency Hell going on. 
```
Could not retrieve the list of available versions for provider hashicorp/google: no available releases match the given constraints >= 3.43.0, >= 3.50.0, >= 3.53.0, >=
│ 3.64.0, ~> 4.0, >= 4.28.0, < 5.0.0, 5.26.0, < 6.0.0
```

### Terraform Configuration

```hcl
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 4.0"
    }
  }
  required_version = ">= 0.13"
}
```

### Terraform Version

```sh
Terraform v1.8.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.84.0 <- would use >5.0
+ provider registry.terraform.io/hashicorp/google-beta v4.84.0 <- would use >5.0
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/hashicorp/random v3.6.2
+ provider registry.terraform.io/hashicorp/time v0.11.1
```

**Comment by gtsorbo** [comment]
Thanks for your patience. The VM modules have been bumped to ~>11.0 and the ~>4.0 provider constraint in the example has been lifted: https://github.com/terraform-google-modules/terraform-google-github-actions-runners/pull/152
